modules/v2/main.tf

```diff
@@ -14,6 +14,30 @@
  * limitations under the License.
  */
 
+locals {
+  create_service_account        = var.service_account != null ? false : var.create_service_account
+  service_account               = var.service_account != null ? var.service_account : (var.create_service_account ? google_service_account.sa[0].email : null)
+  service_account_project_roles = local.create_service_account ? var.service_account_project_roles : []
```

Comment:
Is there any reason to have this check? IMO we could let IAM be added for a user created SA too?

Reply:
Updated.